### ISSUE TITLE
Tokens incorrect for template strings

### ISSUE THREAD

**Issue by zpao** [facebook/esprima#101]
See https://gist.github.com/zpao/3225299891281ef785a8 for example. I compared the tokens for a simple template string against what espree is doing. It seems pretty obvious to me that what we have is incorrect.

Should this be filed upstream? Probably. It's way easier to start discussing here and see if I can convince @fkling to fix it properly there :stuck_out_tongue:

**Comment by fkling** [comment]
This is probably the upstream problem? https://github.com/jquery/esprima/issues/1002

**Comment by zpao** [comment]
I knew there was a reason we kept you around :) Thanks for finding. I'll follow along there.

**Comment by fkling** [comment]
:P

**Comment by xjamundx** [comment]
Related to https://github.com/eslint/espree/issues/44 ?

**Comment by mikesherov** [comment]
@xjamundx yes. In fact, I'm working on this at the moment in the harmony branch.

**Comment by mikesherov** [comment]
@fkling @zpao fix incoming in jquery/esprima . The espree patch has problems with nested templates and object literals inside templates. I'll link as soon as it's done.

**Comment by mikesherov** [comment]
https://github.com/jquery/esprima/pull/1104

**Comment by mikesherov** [comment]
@zpao @fkling this is now fixed in esrpima harmony. I have another patch for template strings: https://github.com/jquery/esprima/pull/1124/files which is a bit less serious (unless you're using legacy octal literals in non strict mode template strings)

Let me know if you guys needs help porting the patches over.

**Comment by zpao** [comment]
Merge is #106, thanks!
